src/integrationTest/java/com/mongodb/hibernate/crud/BasicCrudIntegrationTests.java

```diff
@@ -83,6 +85,25 @@ void testSimpleEntityInsertion() {
             assertCollectionContainsExactly(expectedDocument);
         }
 
+        @Test
+        void testDynamicInsert() {
+            sessionFactoryScope.inTransaction(session -> {
+                var book = new BookDynamicallyInserted();
+                book.id = 1;
+                book.title = "War and Peace";
+                book.publishYear = 1867;
+                session.persist(book);
+            });
+            var expectedDocument = BsonDocument.parse(
+                    """
+                    {
+                        _id: 1,
+                        title: "War and Peace",
+                        publishYear: 1867
+                    }""");
+            assertCollectionContainsExactly(expectedDocument);
```

Comment:
`author` won't be inserted for it is null, thus ending up with a more compact storage mechanism.
The troublesome thing is the annotations of `@DynamicInsert` and `@DynamicUpdate` are both deprecated.

Reply:
They are not deprecated. However, it is not clear whether we should support `@DynamicInsert` at the moment or not, so let's not introduce such a test for now.

P.S. I'll reflect the renaming done here in https://github.com/mongodb/mongo-hibernate/pull/64, so that's not lost.